src/Microsoft.AspNet.Mvc.Extensions/ViewDataDictionary.cs

```diff
@@ -426,6 +427,12 @@ private Type GetModelType(object value)
             return (value == null) ? _declaredModelType : value.GetType();
         }
 
+        private static bool IsCompatibleWith([NotNull] Type type, object value)
+        {
+            return (value == null && (!type.GetTypeInfo().IsValueType || Nullable.GetUnderlyingType(type) != null)) ||
+                (value != null && type.GetTypeInfo().IsAssignableFrom(value.GetType().GetTypeInfo()));
```

Comment:
This is a bit of an unreadable mess, can you format this better? 

Using `if` is fine.